src/pudl/output/ferc714.py

```diff
@@ -0,0 +1,400 @@
+"""Functions & classes for compiling derived aspects of the FERC Form 714 data."""
+import pandas as pd
+
+import pudl
+
+
+################################################################################
+# Helper functions
+################################################################################
+def add_dates(rids_ferc714, report_dates):
+    """
+    Broadcast respondent data across dates.
+
+    Args:
+        rids_ferc714 (pandas.DataFrame): A simple FERC 714 Respondent ID dataframe,
+            without any date information.
+        report_dates (ordered collection of datetime): Dates for which each respondent
+            should be given a record.
+
+    """
+    if "report_date" in rids_ferc714.columns:
+        raise ValueError("report_date already present, can't be added again!")
+    # Create DataFrame with all report_date and respondent_id_ferc714 combos
+    dates_rids_df = (
+        pd.DataFrame(
+            index=pd.MultiIndex.from_product([
+                report_dates, rids_ferc714.respondent_id_ferc714.unique()],
+                names=["report_date", "respondent_id_ferc714"]
+            )
+        ).reset_index()
+    )
+    return pd.merge(rids_ferc714, dates_rids_df, on="respondent_id_ferc714")
+
+
+def get_all_utils(pudl_out):
+    """
+    Compile IDs and Names of all known EIA Utilities.
+
+    Grab all EIA utility names and IDs from both the EIA 861 Service Territory table and
+    the EIA 860 Utility entity table. This is a temporary function that's only needed
+    because we haven't integrated the EIA 861 information into the entity harvesting
+    process and PUDL database yet.
+
```

Comment:
What's this pudl_out argument? Looks like not a Path...?

Reply:
Added `Args:` section.